src/TemplateConverter.ump

```diff
@@ -435,16 +461,19 @@ class TemplateConverter {
             return false;
         }
         
-            // Hack
-        int pos = line.indexOf("<%@");
-        
-        for (int i = 0; i < pos; ++i) {
-            if (line.charAt(i) != ' ')
-                return true;
+        // In JET, the newline before a directive will not be included, but Umple will include it
+        // So if the next line is a directive with some spaces before, should not add the newline
+        return !isSpacesFollowedByDirective(line);
+    }
+    
+    // This is used for the directive hack
+    private boolean isSpacesFollowedByDirective(String line) {
+        int pos = line.indexOf(JETConstants.DirectiveStart);
+        if (pos == -1) {
+            return false;
         }
         
-        
-        return pos == -1;
+        return containsOnlySpaces(line.substring(0, pos));
```

Comment:
There are some distinctions made between tabs and spaces for JET (I believe) so for that function it considers them to be separate. Currently however the converter will, for regular text, convert the tabs to spaces anyway, since their output will be different in Umple templates regardless.

As for if it is faster, I couldn't say for sure. Doing trim would probably be slower, since it will be creating a new string object, but it is hard to know for sure.